src/pylhe/__init__.py

```diff
@@ -120,14 +122,30 @@ def loads():
     pass
 
 
-def readLHEInit(thefile):
+def _open_gzip_file(filepath):
```

Comment:
This is going to need work as [`xml.etree.ElementTree.iterparse`](https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.iterparse) seems unhappy at the moment

```
$ file unweighted_events.lhe.gz 
unweighted_events.lhe.gz: gzip compressed data, was "tmp_0_unweighted_events.lhe", last modified: Tue Mar 30 07:22:36 2021, max compression, original size modulo 2^32 15580732
$ python
Python 3.8.6 (default, Jan  5 2021, 00:14:15) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pylhe
>>> pylhe.readLHEInit("unweighted_events.lhe.gz")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/feickert/Code/GitHub/Scikit-HEP/pylhe/src/pylhe/__init__.py", line 148, in readLHEInit
    for event, element in ET.iterparse(filepath, events=["end"]):
  File "/home/feickert/.pyenv/versions/3.8.6/lib/python3.8/xml/etree/ElementTree.py", line 1229, in iterator
    data = source.read(16 * 1024)
  File "/home/feickert/.pyenv/versions/3.8.6/lib/python3.8/gzip.py", line 288, in read
    self._check_not_closed()
  File "/home/feickert/.pyenv/versions/3.8.6/lib/python3.8/_compression.py", line 14, in _check_not_closed
    raise ValueError("I/O operation on closed file")
ValueError: I/O operation on closed file
```